### PR TITLE
[backend] authorize api filters keys for retention policies checking (#5153)

### LINKED ISSUE #5153: Can't add api filters to retention policies
If we create a retention policy with an api filter, nothing happens after clicking on the 'verify' button --> the 'create' button is not accesible and we can't create the retention policy.

image

### PR DESCRIPTION
**Pull request by Archidoit**
review by the Filters team

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/utils/filtering/filtering-utils.ts [hunk 1]
```diff
@@ -222,16 +222,16 @@ const getConvertedRelationsNames = (relationNames: string[]) => {
  * - convert relation refs key if any
  * @param filterGroup
  */
-export const checkAndConvertFilters = (filterGroup?: FilterGroup | null) => {
+export const checkAndConvertFilters = (filterGroup?: FilterGroup | null, noFiltersChecking = false, authorizeNestedFiltersKeys = false) => {
```
Comment (labo-flg) [inline opencti-platform/opencti-graphql/src/utils/filtering/filtering-utils.ts]: Please prefer using an `opts` parameter with different options inside, so we don't expand the arguments lists infinitely :)

Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/utils/filtering/filtering-utils.ts]: yes !!

Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/utils/filtering/filtering-utils.ts]: fixed

**Review by labo-flg — APPROVED** [review]
Tested on dashboards, relationship filter by marking, triggers, and of course retention policies. It seems ok to me.

**Pull request merged**